### DISCUSSION TITLE
hit_sphere return value

### DISCUSSION THREAD

**Discussion by crypto-dot** [RayTracing/raytracing.github.io#954] [Q&A] [answered]
The hit_sphere functions returns (-b - sqrt(discriminate))/ (2a). Is there any reason as to why we are returning the difference between -b and the sqrt(discriminate) versus the addition of -b and the sqrt(discriminate) since quadratic functions can have 2 roots.  I tested the difference in my code and the one adding those two values together produced a sphere with much vibrant colors. Is this a personal design decision? Or is there a mathematical reason for this choice?

```
double hit_sphere(const point3& center, double radius, const ray& r) {
    vec3 oc = r.origin() - center;
    auto a = dot(r.direction(), r.direction());
    auto b = 2.0 * dot(oc, r.direction());
    auto c = dot(oc, oc) - radius*radius;
    auto discriminant = b*b - 4*a*c;
    if (discriminant < 0) {
        return -1.0;
    } else {
        return (-b - sqrt(discriminant) ) / (2.0*a);
    }
}

```

**Comment by hollasch** [comment, marked as answer]
At that point in the code, the scene and camera are pretty much hard-coded. All objects are in front of the camera. Given this, the nearest root is guaranteed to be the one with the smallest ray _t_ value. Also, note the comment:

> If you change your sphere center to $z = +1$ you will get exactly the same picture because you see the things behind you. This is not a feature! We’ll fix those issues next.

Also note the latter comment:

> We'll just assume the closest hit point (smallest $t$).
